src/components/Dashboard/ThisWeek/WeeklyActivity/WeeklyActivity.test.tsx

```diff
@@ -135,4 +144,32 @@ describe('WeeklyActivity', () => {
     );
     expect(await findByLabelText('Weekly Report')).toBeInTheDocument();
   });
+
+  it('should not show Fill out weekly report button', async () => {
+    const { queryByRole, findByRole } = render(
+      <SnackbarProvider>
+        <ThemeProvider theme={theme}>
+          <MockedProvider
+            mocks={[
+              ...GetWeeklyActivityQueryDefaultMocks(),
+              ...CurrentCoachingAnswerSetDefaultMocks(),
+            ]}
+            addTypename={false}
+          >
+            <WeeklyActivity accountListId={accountListId} />
+          </MockedProvider>
+        </ThemeProvider>
+      </SnackbarProvider>,
+    );
+
+    expect(
+      await findByRole('link', { name: 'View Activity Detail' }),
+    ).toBeInTheDocument();
+
+    await waitFor(() => {
```

Comment:
Why is this test passing when we give it the same mocked data as the other test where the button is showing?

Reply:
Thanks for figuring that out!